x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/servicenow/servicenow_params.tsx

```diff
@@ -13,90 +13,99 @@ import {
+      'xpack.triggersActionsUI.components.builtinActionTypes.servicenow.severitySelectLawOptionLabel',
+      { defaultMessage: 'Low' }
+    ),
+  },
+];
 
 const ServiceNowParamsFields: React.FunctionComponent<
   ActionParamsProps<ServiceNowActionParams>
 > = ({ actionParams, editAction, index, errors, messageVariables }) => {
-  const { title, description, comment, severity, urgency, impact, savedObjectId } =
-    actionParams.subActionParams || {};
-
-  const isActionBeingConfiguredByAnAlert = messageVariables
-    ? isSome(extractActionVariable(messageVariables, AlertProvidedActionVariables.alertId))
-    : false;
-
-  const selectOptions = [
-    {
-      value: '1',
-      text: i18n.translate(
-        'xpack.triggersActionsUI.components.builtinActionTypes.servicenow.severitySelectHighOptionLabel',
-        {
-          defaultMessage: 'High',
-        }
-      ),
-    },
-    {
-      value: '2',
-      text: i18n.translate(
-        'xpack.triggersActionsUI.components.builtinActionTypes.servicenow.severitySelectMediumOptionLabel',
-        {
-          defaultMessage: 'Medium',
-        }
-      ),
-    },
-    {
-      value: '3',
-      text: i18n.translate(
-        'xpack.triggersActionsUI.components.builtinActionTypes.servicenow.severitySelectLawOptionLabel',
-        {
-          defaultMessage: 'Low',
-        }
-      ),
+  const {
+    // short_description is a servicenow property
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    incident: { short_description, description, severity, urgency, impact },
+    comments,
+  } = useMemo(() => {
+    return (
+      actionParams.subActionParams ?? {
+        incident: {
+          short_description: null,
+          description: null,
+          severity: null,
+          urgency: null,
+          impact: null,
+        },
+        comments: [],
+      }
+    );
+  }, [actionParams.subActionParams]);
+  const editSubActionProperty = useCallback(
+    (key: string, value: any) => {
+      const newProps =
+        key !== 'comments'
+          ? {
+              ...actionParams.subActionParams,
+              incident: {
+                ...(actionParams.subActionParams ? actionParams.subActionParams.incident : {}),
+                [key]: value,
+              },
+            }
+          : { ...actionParams.subActionParams, [key]: value };
+      editAction('subActionParams', newProps, index);
     },
-  ];
-
-  const editSubActionProperty = (key: string, value: {}) => {
-    const newProps = { ...actionParams.subActionParams, [key]: value };
-    editAction('subActionParams', newProps, index);
-  };
+    [actionParams.subActionParams, editAction, index]
+  );
 
   useEffect(() => {
     if (!actionParams.subAction) {
       editAction('subAction', 'pushToService', index);
     }
-    if (!savedObjectId && isActionBeingConfiguredByAnAlert) {
-      editSubActionProperty('savedObjectId', `${AlertProvidedActionVariables.alertId}`);
-    }
-    if (!urgency) {
```

Comment:
@YulNaumenko I removed this automatic setting because i can see in the type that these params can be null. If product says we always need to have these set, we should make it required in the type.